### PR TITLE
Run tests/integration.rs Actions CI test.yml

### PR DESCRIPTION
**Pull request by DanGould**

### CODE REVIEW

**Review by DanGould — COMMENTED** [review]
use docker-compose v2.10.2

**Comment by DanGould** [comment]
I think this test is failing because the checks are done synchrronously. If we do polling checks with timeouts the problems should go away

**Comment by nickfarrow** [comment]
Adding 60s timeout didn't help 😅, havent tried polling yet

**Comment by DanGould** [comment]
it runs! but there are some infinite loops. The test itself would benefit from a timeout on the test() thread

File: tests/integration.rs [hunk 1]
```diff
@@ -13,63 +14,74 @@ mod integration {
...
-        Command::new("docker")
-            .arg("cp")
-            .arg("compose-merchant_lnd-1:/data/chain/bitcoin/regtest/admin.macaroon")
-            .arg(format!("{}/merchant-admin.macaroon", &tmp_path))
-            .output()
-            .expect("failed to copy admin.macaroon");
-        println!("copied merchant-admin.macaroon");
+        let bitcoin_rpc = loop {
+            if let Ok(btcrpc) = do_event() {
+                if btcrpc.get_best_block_hash().is_ok() {
```
Comment (nickfarrow) [inline tests/integration.rs:44]: Infinite loop got me here, changing to 
```
match btcrpc.get_best_block_hash() {
    Ok(_) => break btcrpc,
    Err(e) => println!("Attempting to contact btcrpc: {}", e),
}
```
would be informative of where the test is breaking

File: tests/integration.rs [hunk 2]
```diff
@@ -13,63 +14,74 @@ mod integration {
...
+                .output()
+                .expect("failed to copy admin.macaroon");
+            println!("copied merchant-admin.macaroon");
+
+            // Connecting to LND requires only address, cert file, and macaroon file
+            let client =
+                tonic_lnd::connect(address_str, &cert_file, &macaroon_file).await;
+                
+            if let Ok(mut client) = client {
+                if client.get_info(tonic_lnd::rpc::GetInfoRequest {}).await.is_ok() {
```
Comment (nickfarrow) [inline tests/integration.rs:80]: and here
```
match client.get_info(tonic_lnd::rpc::GetInfoRequest {}).await {
    Ok(_) => break client,
    Err(e) => println!("Attempting to contact lnd: {}", e),
}
```

**Comment by nickfarrow** [comment]
Works for me now, great code!

**Comment by DanGould** [comment]
Took @nickfarrow suggestions and removed infinite loops

**Comment by DanGould** [comment]
followed my own advice and got rid of formatting errors for non-formatting commits

**Pull request merged**